autosklearn/data/target_validator.py

```diff
@@ -175,11 +180,27 @@ def _fit(
         assert self.encoder is not None
 
         # remove ravel warning from pandas Series
-        if ndim > 1 and np.shape(y_train)[1] == 1 and hasattr(y_train, "to_numpy"):
-            y_train = typing.cast(pd.DataFrame, y_train)
-            self.encoder.fit(y_train.to_numpy().ravel())
-        else:
+        if ndim > 1:
             self.encoder.fit(y_train)
+        else:
+            if hasattr(y_train, 'iloc'):
+                y_train = typing.cast(pd.DataFrame, y_train)
+                self.encoder.fit(y_train.to_numpy().reshape(-1, 1))
+            else:
+                self.encoder.fit(np.array(y_train).reshape(-1, 1))
+
+        # we leave objects unchanged, so no need to store dtype in this case
+        if hasattr(y_train, 'dtype'):
+            # Series and numpy arrays are checked here
+            # Cast is as numpy for mypy checks
+            y_train = typing.cast(np.ndarray, y_train)
+            if is_numeric_dtype(y_train.dtype):
+                self.dtype = y_train.dtype
+        elif hasattr(y_train, 'dtypes') and is_numeric_dtype(typing.cast(pd.DataFrame,
+                                                                         y_train).dtypes[0]):
+            # This case is for pandas array with a single column
```

Comment:
Don't you have to also check the dimensionality of the pandas array here if you assume it only has a single column?

Reply:
This comment tries to clarify [[this](https://github.com/automl/auto-sklearn/blob/61bdaee47f7fdb593569e5d638b672519300a00c/autosklearn/data/target_validator.py#L192)  in that we only care about numerical dtype. If no dtype exists, then we do not mind the else. We do so for numerical so the inverse transform type is not type object.